src/vision/status_quo/alan_has_an_event_loop.md

```diff
@@ -0,0 +1,104 @@
+
+## The story
+
+As a first Rust Project, Alan decides to program his own IRC Client.
+
+Since it is Alans first Project in Rust, it is going to be a private one. He is going to use it on is Mac, so he decides to go with the cocoa crate to not have to learn any Framework specific quirks. This way Alan can get a feel of Rust itself.
+
+### Alans hopes and dreams
+Despite a learning curve, he managed to creating a first window and have some buttons and menus works. After the initialisation is done, the App hand over control to [CFRunLoop::Run](https://developer.apple.com/documentation/corefoundation/1542011-cfrunlooprun?language=occ).
+
+Once Alan is happy whit his Mock UI, he wants to hook it up with some actual features. He is happy to learn that Rust has Features he already knows.
+* Promises => Futures
+* Observables => Streams.
+
+Alan smiles, thinking he knows what and more importantly how to do this.
+
+### First time dealing with runtimes
+
+Unfortunately, coming from frameworks like Angular or Node.js, Alan is not used that he himself is responsible for driving the progress of Async/Streams. 
+
+After reading up about Runtimes, his mental image of a runtime is something like: 
+
+```rust
+impl Runtime {
+    fn run() {
+        while !self.tasks.is_empty() {
+            while let Some(task) = self.awoken_tasks.pop() {
+                task.poll();
+                //... remove finished task from 'tasks'
+            }
+        }
+    }
+}
+```
+
+Coming from Single-Threaded Angular development, Alan decides to limit his new App to Single-Threaded. He does not feel like learning about Send/Sync/Mutex as well as struggling with the borrow checker.
+
+On top of that, his App is not doing any heavy calculation so he feels async should be enough to not block the main thread too bad and have a hanging UI.
+
+### Fun time is over
+
+Soon Alan realises that he cannot use any of those runtimes because they all take control of the thread and block. The same as the OS Event loop.
+
+Alan spends quite some time to look through several runtime implementations. Ignoring most internal things, all he wants is a runtime that looks a bit like this:
+
+```rust
+impl Runtime {
+    fn make_progress() {
+        while let Some(task) = self.awoken_tasks.pop() {
+            task.poll();
+            //... remove finished task from 'tasks'
+        }
+    }
+    fn run() {
+        while !self.tasks.is_empty() {
+            self.make_progress();
+        }
+    }
+}
+```
+
+It could be soo easy. Unfortunately he does not find any such solution. Having already looked through quite a bit of low level documentation and runtime code, Alan thinks about implementing his own runtime...
```

Comment:
```suggestion
It could be so easy. Unfortunately he does not find any such solution. Having already looked through quite a bit of low level documentation and runtime code, Alan thinks about implementing his own runtime...
```